org.lflang/src/org/lflang/generator/c/CGenerator.java

```diff
@@ -416,9 +416,8 @@ public void doGenerate(Resource resource, LFGeneratorContext context) {
       copyTargetFiles();
       generateHeaders();
       code.writeToFile(targetFile);
-    } catch (IOException e) {
-      //noinspection ThrowableNotThrown,ResultOfMethodCallIgnored
-      Exceptions.sneakyThrow(e);
+    } catch (Exception e) {
+      errorReporter.reportError(e.getMessage());
```

Comment:
This is the line that is hit by the example though. Should we add a `reportError` method that allows passing in an exception so that we can preserve the stack trace and print it later?

Reply:
OK, maybe we should handle the stack trace here too.

Why not just print the stack trace immediately? We should never have to print a stack trace anyway except when something else is already horribly wrong.

I also just noticed that this catches a general `Exception` instead of an `IOException`, which seems unnecessary.

Reply:
Oh, now I remember why I was OK with this. This is actually not a situation when something is horribly wrong. This line can be reached when the user provides an invalid path to the TargetConfig. In that case, probably just the error message will usually be sufficient to figure out what the problem is, _and_ it isn't really the compiler's fault, so maybe the compiler shouldn't have to print out an ugly stack trace.

Reply:
If we only catch an `IOException` as you suggest, the example that Edward provides in #1785 is no longer handled correctly.

Reply:
I suppose we can catch the runtime exception later, but we weren't doing that either...

Reply:
Ah, okay. Then since we are catching more general errors such as NPEs here, I guess I agree with you that we should do something with the stack trace.

Reply:
Reverting this to a catch for `IOException` and handling runtime exceptions further up the call stack might be better though...

Reply:
That also makes sense. I am happy as long as we never silence exceptions by catching them without handling them properly. If we forget to handle some exceptions that "should be impossible", then we will still get a stack trace, which I would be satisfied with.

Reply:
OK, narrowed it down and added a separate catch for `RuntimeException`.